### DISCUSSION TITLE
How to get detail table entityid in AfterSave method of master

### DISCUSSION THREAD

**Discussion by reach2rv** [serenity-is/Serenity#5360] [Help] [answered]
in AfterSave method of Master repository in Masterdetail Form how to get entityid of details table. when I check current row it shows master entityid but details entityid is null

#4370

**Comment by minhhungit** [comment, marked as answer]
Haa, I've just seen this post, I posted solution on the https://github.com/serenity-is/Serenity/issues/4370 some minutes ago
```csharp
protected override void AfterSave()
{
    base.AfterSave();

    var products = this.Connection.List<Entities.OrderProductRow>(new Criteria(Entities.OrderProductRow.Fields.OrderId) == (int)Row.Id);
}
```

**Reply by reach2rv** [reply]
thank you @minhhungit for your quick reply.. it works as expected
